### PR TITLE
Protect against custom properties

### LINKED ISSUE #624: An error is occured when lint @apply rules
I tried to lint CSS code that uses [@apply rules](http://tabatkins.github.io/specs/css-apply-rule/) by [postcss-apply](https://github.com/pascalduez/postcss-apply), but the following error is occured.

```
Message:
    Cannot read property '0' of null
```

and here is my CSS code.

```css
:root {
  --truncate: {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}
```

I wish your help. Thanks.

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #624

@davidtheclark Is this the best way to protect against this?

### CODE REVIEW

File: src/rules/selector-no-vendor-prefix/index.js [hunk 1]
```diff
@@ -22,11 +22,11 @@ export default function (actual) {
 
       // Check each pseudo-selector
       styleSearch({ source: selector, target: ":" }, match => {
-        const pseudoSelector = /:{1,2}[a-z-]+\b/.exec(selector.slice(match.startIndex))[0]
-
-        if (isAutoprefixable.selector(pseudoSelector)) {
+        const pseudoSelector = /:{1,2}[a-z-]+\b/.exec(selector.slice(match.startIndex))
+        if (pseudoSelector === null) { return }
+        if (isAutoprefixable.selector(pseudoSelector[0])) {
```
Comment (davidtheclark) [inline src/rules/selector-no-vendor-prefix/index.js]: This is an improvement, yes. And if it fixes the problem and all tests still pass, go with it.

**Comment by davidtheclark** [comment]
Don't forget to update the changelog please.

**Comment by jeddy3** [comment]
@davidtheclark Is this a patch or a feature?

**Comment by davidtheclark** [comment]
Oh, I'd consider it a patch.

**Pull request merged**